vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java

```diff
@@ -27,4 +39,52 @@ PKCS10CertificationRequest getBcCsr() {
     public X500Principal getSubject() {
         return new X500Principal(csr.getSubject().toString());
     }
+
+    public List<String> getSubjectAlternativeNames() {
+        return getExtensions()
+                .map(extensions -> GeneralNames.fromExtensions(extensions, Extension.subjectAlternativeName))
+                .filter(Objects::nonNull)
+                .flatMap(generalNames -> Arrays.stream(generalNames.getNames()))
+                .map(Pkcs10Csr::toString)
+                .collect(toList());
+    }
+
+    /**
+     * @return true if CA certificate
+     */
+    public boolean getBasicConstraints() {
+        return getExtensions()
+                .map(BasicConstraints::fromExtensions)
+                .findAny()
```

Comment:
How/why does this work? Does `getExtension()` always return 1 element or does `isCA()` return true for all of them?

Reply:
I found an answer here: https://access.redhat.com/documentation/en-US/Red_Hat_Certificate_System/8.0/html/Admin_Guide/Standard_X.509_v3_Certificate_Extensions.html
> ... , certificate extensions appear in sequence and only one instance of a particular extension may appear per certificate; for example, a certificate may contain only one subject key identifier extension.

I have no idea why the BC API allows multiple instances of the same extension...

Reply:
Shouldn't `getExtensions()` return `Extensions` or `Optional<Extensions>`?

Or at least consider
```
    public boolean getBasicConstraints() {
        return getExtensions()
                .map(BasicConstraints::fromExtensions)
                .anyMatch(BasicConstraints::isCA);
    }
```

Reply:
Good point. It should be `OptionaBoolean`.